src/models/connectionStore.ts

```diff
@@ -202,38 +199,29 @@ export class ConnectionStore {
 	 * @param {forceWritePlaintextPassword} whether the plaintext password should be written to the settings file
 	 * @returns {Promise<IConnectionProfile>} a Promise that returns the original profile, for help in chaining calls
 	 */
-	public saveProfile(profile: IConnectionProfile, forceWritePlaintextPassword?: boolean): Promise<IConnectionProfile> {
+	public async saveProfile(profile: IConnectionProfile, forceWritePlaintextPassword?: boolean): Promise<IConnectionProfile> {
 		const self = this;
-		return new Promise<IConnectionProfile>((resolve, reject) => {
-			// Add the profile to the saved list, taking care to clear out the password field if necessary
-			let savedProfile: IConnectionProfile;
-			if (profile.authenticationType === Utils.authTypeToString(AuthenticationTypes.AzureMFA)) {
-				savedProfile = Object.assign({}, profile, { azureAccountToken: '' });
+		// Add the profile to the saved list, taking care to clear out the password field if necessary
+		let savedProfile: IConnectionProfile;
+		if (profile.authenticationType === Utils.authTypeToString(AuthenticationTypes.AzureMFA)) {
+			savedProfile = Object.assign({}, profile, { azureAccountToken: '' });
+		} else {
+			if (forceWritePlaintextPassword) {
+				savedProfile = Object.assign({}, profile);
 			} else {
-				if (forceWritePlaintextPassword) {
-					savedProfile = Object.assign({}, profile);
-				} else {
-					savedProfile = Object.assign({}, profile, { password: '' });
-				}
+				savedProfile = Object.assign({}, profile, { password: '' });
 			}
+		}
 
-
-			self._connectionConfig.addConnection(savedProfile)
-				.then(() => {
-					// Only save if we successfully added the profile
-					return self.saveProfilePasswordIfNeeded(profile);
-					// And resolve / reject at the end of the process
-				}, err => {
-					reject(err);
-				}).then(resolved => {
-					// Add necessary default properties before returning
-					// this is needed to support immediate connections
-					ConnInfo.fixupConnectionCredentials(profile);
-					resolve(profile);
-				}, err => {
-					reject(err);
-				});
-		});
+		await self._connectionConfig.addConnection(savedProfile)
```

Comment:
nit: Another opportunity for clean (especially the catch which just catches and immediately rethrows...)